pkg/prometheus/agent/operator.go

```diff
@@ -1184,3 +1187,25 @@ func keyToDaemonSetKey(p monitoringv1.PrometheusInterface, key string) string {
 	keyParts := strings.Split(key, "/")
 	return fmt.Sprintf("%s/%s", keyParts[0], fmt.Sprintf("%s-%s", prompkg.Prefix(p), keyParts[1]))
 }
+
+// a fallback CEL validation function for older kubernetes versions when CEL is not available.
+func validateDaemonSetModeSpec(p *monitoringv1alpha1.PrometheusAgent) error {
```

Comment:
Something I just realized: 
The ScrapeConfig is not a great fit for the DaemonSet scenario. I think it might be better to deny the DaemonSet from being configured with a ScrapeConfig until we know what to do with it? 

For context: The ScrapeConfig lets you configure the entire behavior of the Prometheus service discovery. As for Daemonset, it operates strictly on the K8s SD, and with specific field selectors. Allowing the users configuration freedom with the ScrapeConfig will defeat the purpose of the DaemonSet.